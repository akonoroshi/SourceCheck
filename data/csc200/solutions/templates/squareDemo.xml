<project name="square" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1" guid="e763d042-647f-4279-a51d-27a0a499101b" assignment="squareDemo"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAAFg0lEQVR4Xu2dTUhtZRSGlyLHSgnBI5iDBhGSdyCSDhTD3wzMSSg6cCRoAxWcNEhF7yQJhHQgoYgg6ODODJxHiEo60Ab+YtwwGjiQZkn+a3wbrnU6N64n4fqutd4NDjx+++z3Xc/DPupgf2m3t7e3woMTeKQJpFHAR5o8LxtNgAJShEedwEsFPDs7k4WFBTk+Pn6t4QoKCqS1tVXS09Nf63V5scebQJKAMzMzUltb+6BEp/tL8vPcoFzd89fLeDwuHzz9Xk5PT6Prbm1tSXNz84My8GQdE0gQ8ObmRg4ODh6c/I9v6uVJPCOl98ke/U329/fvzikqKkrpfC7WOYEEAScmJqSmpuauSXt7u0xOTkp1dbUsLi7K6OioTE1NSVtbm+Tm5kbr6urqJBaLSUZGhjQ1NUWvvT3zqcTf+vtjNPvr53Iy+L68+fkzufjh22jN9S8/JkwsCBjufC+O4uJinRNl6pQmkCDgwMCAtLS03L1BR0eH7O7uRt+vra1JSUmJbG5uyuHhoYSP6r6+PmloaJDt7W3p7u6W2dnZaG187rMkAW9+/zX62Z/jH0vGh81y9dN3SQJubGzcvVZWVpZSES7WOYEEAefm5qSwsPCuSVdXVyRaZWWlLC0tRQLu7OxIT0+P5OXlydDQkMzPz0tnZ2ck4PT0dHTuO89aEwSMffKFxD7qlJOnTyTc6U6+fDdpWuH1IPmLo6KiQudEmTqlCST9EbK8vPzKNwiS5ufny3+tfW+hXXLeSO0v2dhXz2V9fT269uXlpdTX178yBxfon0CSgEdHR7K3tydpaWn/u11paank5OTc+/yLiwtZWVmJ1l9dXUV33Ozs7Hufz4V6J8B/ROtlZyI5BTSBUW8JCqiXnYnkFNAERr0lKKBediaSU0ATGPWWoIB62ZlITgFNYNRbggLqZWciOQU0gVFvCQqol52J5BTQBEa9JSigXnYmklNAExj1lqCAetmZSE4BTWDUW4IC6mVnIjkFNIFRbwkKqJedieQU0ARGvSUooF52JpJTQBMY9ZaggHrZmUhOAU1g1FuCAuplZyI5BTSBUW8JCqiXnYnkFNAERr0lKKBediaSU0ATGPWWoIB62ZlITgFNYNRbggLqZWciOZyA4YHn4Un84Yn94cn7PGxPAE7AMO7BwcFoz5DwuN6srCypqqqKHozOw94EIAUMY+7t7ZXwvOp/HtfX1xK+hoeHpby83B4Nh41gBezv73/prk1ByrCfSWNjo2RmZjpEZqsypIBjY2OyuroaTfr8/Dy624V9SMLeJDxsTQBOwLBd18jIiIyPj0vYPZM7Z9oS7t9t4AS0PW62o4B0AGoCvANC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7VmAJC4fAXhgL6Yw7V+C++fLi3LS0M4AAAAABJRU5ErkJggg==</thumbnail><stage name="Stage" guid="e763d042-647f-4279-a51d-27a0a499101b" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" ternary="true" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><watcher var="length" style="normal" x="10" y="10" color="243,118,29"/><sprite name="Sprite" idx="1" x="-100.72688120053613" y="38.91926633347617" heading="90" scale="1" rotation="1" draggable="true" costume="0" color="80,80,80" pen="tip" id="9"><costumes><list id="10"></list></costumes><sounds><list id="11"></list></sounds><blocks></blocks><variables></variables><scripts><script x="148.00000000000003" y="53"><block id="125" s="receiveGo"></block><block id="178" s="down"></block><block id="246" s="doSetVar"><l>length</l><block id="292" s="reportRandom"><l>50</l><l>100</l></block></block><block id="212" s="doRepeat"><l>4</l><script><block id="213" s="forward"><block id="294" var="length"/></block><block id="214" s="turn"><l>90</l></block></script></block><block id="293" s="up"></block></script></scripts></sprite></sprites></stage><hidden></hidden><headers></headers><code></code><blocks></blocks><variables><variable name="length"><l>60</l></variable></variables><editing></editing></project>