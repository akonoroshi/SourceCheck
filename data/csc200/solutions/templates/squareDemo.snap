snapshot {
    @defHint SET_VAR {
        :`We need to store the size of the square in a variable, which tells us how long each side should be.`
    }
    @defHint RANDOM {
        :`The square should be a different, random size each time it is drawn.`
    }
    @defHint VAR {
        :`Each side of the square should be the same, random length that is stored in the variable.`
    }
    stage:Stage {
        sprite:Sprite {
            script {
                receiveGo
                down
                @hint SET_VAR @hint var
                doSetVar(varMenu:length, @hint RANDOM reportRandom(literal:50, literal:100))
                doRepeat(literal:4, script {
                    forward(@hint VAR var:length)
                    turn(literal:90)
                })
                up
            }
        }
    }
    varDec:length
}
